# Sarah Quesnelle, 53914140


### Question 1:

- **Has there been a significant change in average Temperature for BC, Canada between the years of 1917 to 2017?**

### Question 2:

- **Q2: Which Season has seen the greatest change in Temperature over the given years? Winter(January) or Summer(August)**

### Question 3:

- **Has there been a significant change in the amount of snowfall in Canada since 1917?**

In [232]:
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np
import seaborn as sns
import math
import sys
import os
from pathlib import Path
sys.path.append("./code/")
import Project_Functions as ProFun
import glob
pd.options.display.float_format = '{:.1f}'.format

## Testing my Data Cleaning function

In [233]:
test_path = '../data/raw/Canada/csv_canada_1923_2.CSV'
testdf = ProFun.DataClean(test_path)
testdf.head()

Year  Month     Stn_Name Prov  Tm     S     P
0  1923      2   BEAR CREEK   BC NaN 103.1 140.9
1  1923      2  BEAVER LAKE   BC NaN  65.6 106.5
2  1923      2    CHEMAINUS   BC NaN 103.6 124.9
3  1923      2     COWICHAN   BC 0.0  84.1 100.2
4  1923      2      CROFTON   BC NaN 157.1 175.7

## Data Cleaning step

In [210]:
# merging the files
file_path = os.path.join("../data/raw/Canada/", "csv_canada_*.CSV")

# A list of all joined files is returned
list_of_csv = glob.glob(file_path)

In [211]:
dflist = []
# 
for file in list_of_csv:
    df = ProFun.DataClean(file)
    dflist.append(df)

Candf = pd.concat(dflist)

## Save my Dataframe as a csv

In [188]:
save_path = "../data/processed/Canada/CanTempDf.csv"
Candf.to_csv(save_path, index=False)

## Initial Data Analysis

In [231]:
CanTempDf = pd.read_csv("../data/processed/Canada/CanTempDf.csv")

CanTempDf.head()

Year  Month                   Stn_Name Prov  Tm   S   P
0  1917      1                 BEAR CREEK   BC NaN 104 415
1  1917      1                BEAVER LAKE   BC NaN  43 122
2  1917      1                   COWICHAN   BC   1  55 115
3  1917      1  COWICHAN BAY CHERRY POINT   BC   1  24  88
4  1917      1            GOLDSTREAM LAKE   BC NaN 109 224

In [212]:
CanTempDf['Prov'].unique()

array(['BC', 'YT', 'NT', 'AB', 'SK', 'MB', 'ON', 'QC', 'NB', 'NS', 'PE',
       'NL', 'NU', 'XX'], dtype=object)

In [213]:
CanTempDf['Month'].unique()

array([ 1, 10, 11, 12,  2,  3,  4,  5,  6,  7,  8,  9], dtype=int64)

In [230]:
CanTempDf['Year'].unique()

array([1917, 1918, 1919, 1920, 1921, 1922, 1923, 1924, 1925, 1926, 1927,
       1928, 1929, 1930, 1931, 1932, 1933, 1934, 1935, 1936, 1937, 1938,
       1939, 1940, 1941, 1942, 1943, 1944, 1945, 1946, 1947, 1948, 1949,
       1950, 1951, 1952, 1953, 1954, 1955, 1956, 1957, 1958, 1959, 1960,
       1961, 1962, 1963, 1964, 1965, 1966, 1967, 1968, 1969, 1970, 1971,
       1972, 1973, 1974, 1975, 1976, 1977, 1978, 1979, 1980, 1981, 1982,
       1983, 1984, 1985, 1986, 1987, 1988, 1989, 1990, 1991, 1992, 1993,
       1994, 1995, 1996, 1997, 1998, 1999, 2000, 2001, 2002, 2003, 2004,
       2005, 2006, 2007, 2008, 2009, 2010, 2011, 2012, 2013, 2014, 2015,
       2016, 2017], dtype=int64)

In [234]:
CanTempDf.describe(include=np.number)

Year     Month        Tm         S         P
count 1835898.0 1835898.0 1521092.0 1666018.0 1775950.0
mean     1975.8       6.5       3.8      14.4      69.6
std        24.4       3.4      11.4      25.6      67.6
min      1917.0       1.0     -50.0      -3.0      -1.0
25%      1960.0       4.0      -4.3       0.0      25.4
50%      1978.0       7.0       5.6       0.8      54.4
75%      1995.0       9.0      13.3      19.8      92.8
max      2017.0      12.0      27.9     725.9     999.3

In [216]:
CanTempDf.describe(exclude=np.number)

Stn_Name     Prov
count     1835898  1835898
unique       7651       14
top     WOODSTOCK       BC
freq         2298   387119

In [235]:
CanTempDf[CanTempDf['Stn_Name'] == 'WOODSTOCK'].groupby('Prov').count().reset_index()

Prov  Year  Month  Stn_Name    Tm     S     P
0   NB  1096   1096      1096  1085  1090  1089
1   ON  1202   1202      1202  1199  1201  1200

## Observations on Initial Analysis

- There are 13 Provinces in this dataset, 12 of which are valid Canadian Provinces. The 13th Province is labelled as "XX". I am assuming this is a place holder for temperature data that was not assigned a Province value.
- There are 12 unique months of each year given in this dataset. January = 1, February = 2, ..., December = 12.
-  There are 101 unique years ranging from 1917 to 2017.
- There are 7651 unique Stations scattered around Canada. The most popular station is Woodstock. Woodstock, Ontario has had 1202 temperature readings within the given years and Woodstock, New Brunswick has had 1096 temperature readings. The most popular Province is BC.
- 
- 
- 

## Exploratory Data Analysis